src/java/org/apache/cassandra/db/CounterMutation.java

```diff
@@ -164,29 +189,47 @@ public void apply()
         applyCounterMutation();
     }
 
+    private int getNrLocks(Iterable<Lock> locks)
```

Comment:
I'm lost

Reply:
@jakubzytka Can you elaborate? How do you think the locks should be count? To my understanding this should produce the same result as the original code in BDP.

Reply:
We talked it over in slack;

Let's fix the duplicate counting for now.

I'd love to have a debug-only assertion that the iterable is indeed sorted, but I don't think that C* has a mechanism for that.

Reply:
It's already fixed to not count duplicates. It should be sorted according to [`Striped.bulkGet`](https://guava.dev/releases/20.0/api/docs/com/google/common/util/concurrent/Striped.html#bulkGet-java.lang.Iterable-).

Reply:
I will see if I can come up with a test, which will be broken if the iterable is not sorted.

Reply:
duplicate counting does not work;
the `prev` object is not set correctly.

Reply:
Yes, you are completely right! It is good that I am working on bringing tests, however, I am still figuring out how to work with histograms.